mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h

```diff
@@ -2333,49 +2386,44 @@ struct Conv2DRewritePattern : public OpRewritePattern<T> {
           targetGemmDim1Attr.push_back(
               b.getNamedAttr("upper_layer_names",
                              b.getArrayAttr({b.getStringAttr(gemmKPad_name)})));
-        } else if (arg1TargetLayoutName2 == "gemmK") {
+
+          if (convOpType == miopen::ConvOpType::Conv2DOpType) {
```

Comment:
please add comments here on the rationale assigning dimensions to OOB check dimensions, like what you did above for filter.